spec/unit/provider/service/insserv_service_spec.rb

```diff
@@ -36,7 +36,7 @@
   describe "load_current_resource" do
     describe "when startup links exist" do
       before do
-        allow(Dir).to receive(:glob).with("/etc/rc**/S*initgrediant").and_return(["/etc/rc5.d/S18initgrediant", "/etc/rc2.d/S18initgrediant", "/etc/rc4.d/S18initgrediant", "/etc/rc3.d/S18initgrediant"])
+        allow(Dir).to receive(:glob).with("/etc/rc**/**/S*initgrediant").and_return(["/etc/rc5.d/S18initgrediant", "/etc/rc2.d/S18initgrediant", "/etc/rc4.d/S18initgrediant", "/etc/rc3.d/S18initgrediant", "/etc/rc.d/rc3.d/S18initgrediant"])
```

Comment:
this is stubbing a result which can never happen though isn't it?  you should return either strictly the 3-path RHEL version or the 4-path SLES version?  (which may mean that you have to go through all the examples with Dir.glob returning 3-path values and then with Dir.glob returning 4-path values.

Reply:
Thank you @lamont-granquist. I've added the changes.